zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityHighRequestCountIT.java

```diff
@@ -0,0 +1,450 @@
+    ZEEBE
+        .withGatewayConfig(
+            gateway -> {
+              gateway.getMultiTenancy().setEnabled(true);
+              gateway.getExperimental().getIdentityRequest().setEnabled(true);
+              gateway.getSecurity().getAuthentication().setMode(AuthMode.IDENTITY);
+            })
+        .withProperty(
+            "camunda.identity.baseUrl",
+            "http://%s:%d".formatted(IDENTITY.getHost(), IDENTITY.getMappedPort(8080)))
+        .withProperty(
+            "camunda.identity.issuerBackendUrl",
+            "http://%s:%d%s"
+                .formatted(
+                    KEYCLOAK.getHost(), KEYCLOAK.getMappedPort(8080), KEYCLOAK_PATH_CAMUNDA_REALM))
+        .withProperty("camunda.identity.audience", ZEEBE_CLIENT_AUDIENCE)
+        .withRecordingExporter(true)
+        .start()
+        .await(TestHealthProbe.READY);
+
+    associateTenantsWithClient(List.of(DEFAULT_TENANT, TENANT_A), ZEEBE_CLIENT_ID_TENANT_A);
+    associateTenantsWithClient(List.of(DEFAULT_TENANT, TENANT_B), ZEEBE_CLIENT_ID_TENANT_B);
+    associateTenantsWithClient(
+        List.of(DEFAULT_TENANT, TENANT_A, TENANT_B), ZEEBE_CLIENT_ID_TENANT_A_AND_B);
+  }
+
+  @BeforeEach
+  void setup() {
+    RecordingExporter.reset();
+    processId = Strings.newRandomValidBpmnId();
+    process =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .serviceTask("task", b -> b.zeebeJobType("type"))
+            .endEvent()
+            .done();
+  }
+
+  // Regression test: https://github.com/camunda/camunda/issues/23853
+  @Test
+  void shouldHandleHighLoadOnIdentityTenantRequests() throws InterruptedException {
+    // given
+    final long processDefinitionKey;
+    final ThreadFactory factory =
+        Thread.ofVirtual().name("request-factory").uncaughtExceptionHandler((t, e) -> {}).factory();
+    try (final var client = createZeebeClient(ZEEBE_CLIENT_ID_TENANT_A);
+        final ExecutorService executorService = Executors.newThreadPerTaskExecutor(factory); ) {
+      processDefinitionKey =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(process, "process.bpmn")
+              .tenantId(TENANT_A)
+              .send()
+              .join()
+              .getProcesses()
+              .stream()
+              .map(Process::getProcessDefinitionKey)
+              .findFirst()
+              .orElseThrow();
+
+      final List<Callable<ZeebeFuture<ProcessInstanceEvent>>> piList = new ArrayList<>();
+      for (int i = 0; i < 500; i++) {
+        piList.add(
+            () ->
+                client
+                    .newCreateInstanceCommand()
+                    .processDefinitionKey(processDefinitionKey)
+                    .tenantId(TENANT_A)
+                    .send());
+      }
+
+      // when
+      final List<Future<ZeebeFuture<ProcessInstanceEvent>>> results =
+          executorService.invokeAll(piList);
+
+      Thread.sleep(10000);
```

Comment:
:fearful: This looks like flakiness waiting to happen. Can we come up with a different approach?

Reply:
I'll remove it, the `ExecutorService#invokeAll(...)` method actually blocks until all tasks are completed. I missed that.